### PR TITLE
[wasm] Return int from console template

### PR DESCRIPTION
**Pull request by radekdoulik**
Also update the template tests to test that it propagates correctly as
an exit code from the node process.

### CODE REVIEW

File: src/mono/wasm/templates/templates/console/app-support.mjs [hunk 1]
```diff
@@ -166,7 +166,7 @@ try {
             if (runArgs.runtimeArgs.length > 0)
                 INTERNAL.mono_wasm_set_runtime_options(runArgs.runtimeArgs);
 
-            Object.assign(App, { MONO, BINDING, IMPORTS, Module, runArgs });
+            Object.assign(App, { MONO, INTERNAL, BINDING, IMPORTS, Module, runArgs });
```
Comment (radical) [inline src/mono/wasm/templates/templates/console/app-support.mjs:169]: Why is this needed?

**Review by radical — APPROVED** [review]

File: src/mono/wasm/templates/templates/console/app-support.mjs (hunk 1 above)
Comment (radekdoulik) [reply src/mono/wasm/templates/templates/console/app-support.mjs:169]: It is needed here: https://github.com/dotnet/runtime/blob/main/src/mono/wasm/templates/templates/console/app-support.mjs#L49-L66

Without it, we don't call `mono_wasm_exit` and the exit code would be always 0.

**Review by lewing — APPROVED** [review]

File: src/mono/wasm/templates/templates/console/app-support.mjs (hunk 1 above)
Comment (radical) [reply src/mono/wasm/templates/templates/console/app-support.mjs:169]: We should probably remove that `    if (App && App.INTERNAL) {` check anyway. It was useful in `test-main.js`, but not here, IIUC. Can we done in a follow up PR though.
